src/ui/public/agg_types/param_types/field.js

```diff
@@ -26,14 +28,36 @@ export default function FieldAggParamFactory(Private) {
   };
 
   /**
+   * Get the options for this field from the indexPattern
+   */
+  FieldAggParam.prototype.getFieldOptions = function (aggConfig) {
+    const indexPattern = aggConfig.getIndexPattern();
+    let fields = indexPattern.fields.raw;
+
+    fields = fields.filter(f => f.aggregatable);
+
+    if (this.filterFieldTypes) {
+      fields = $filter('fieldType')(fields, this.filterFieldTypes);
+    }
+
+    fields = $filter('orderBy')(fields, ['type', 'name']);
```

Comment:
Yeah, I figured that wasn't intentional. Can't imagine why anyone would want the fields ordered differently based on the existence of `agg.filterFieldTypes`